Error build stlink for Fedora Linux

I have error build stlink 1.4.0 for upcoming Fedora release.
```
/builddir/build/BUILD/stlink-1.4.0/src/tools/gui/stlink-gui.c: In function 'stlink_gui_set_connected':
/builddir/build/BUILD/stlink-1.4.0/src/tools/gui/stlink-gui.c:517:29: error: cast between incompatible function types from 'void (*)(STlinkGUI *)' {aka 'void (*)(struct _STlinkGUI *)'} to 'void * (*)(void *)' [-Werror=cast-function-type]
     g_thread_new ("devmem", (GThreadFunc) stlink_gui_populate_devmem_view, gui);
```

Full build log https://kojipkgs.fedoraproject.org//work/tasks/2219/25032219/build.log
Other components used for build https://kojipkgs.fedoraproject.org//work/tasks/2219/25032219/root.log

It can be because of GCC 8.0.1.

Please help correct this build.

--- a/src/tools/gui/stlink-gui.c
+++ b/src/tools/gui/stlink-gui.c
@@ -224,10 +224,13 @@ stlink_gui_update_devmem_view (STlinkGUI *gui)
 
 
 static void
-stlink_gui_populate_devmem_view (STlinkGUI *gui)
+stlink_gui_populate_devmem_view (gpointer data)
 {
     guint            off;
     stm32_addr_t     addr;
+
+    g_return_if_fail (STLINK_IS_GUI (data));
+    STlinkGUI *gui = (STlinkGUI *)data;
 
     g_return_if_fail (gui != NULL);
     g_return_if_fail (gui->sl != NULL);

